components/buttons/buttons.scss

```diff
@@ -0,0 +1,301 @@
+    }
+
+    &:focus,
+    &.focus {
+      background: linear-gradient(90deg, $alt-blue-2 0%, $clear-blue 100%);
+      box-shadow: inset 0 -0.0625rem 0 0 $alt-blue-3, 0 0 0 0.25rem $rgba-3;
+    }
+  }
+  // Gray Button
+  &--gray {
+    color: $gray-8;
+    background: $white;
+    border-color: $gray-8;
+    bottom: 0;
+    transition: background 0.2s, bottom 0.2s;
+
+    &:hover {
+      color: $white;
+      background: $gray-8;
+      box-shadow: inset 0 -0.0625rem 0 0 $gray-9, 0 0.125rem 0.75rem 0 $rgba-1;
+      bottom: 0.1875rem;
+      transition: background 0.2s, bottom 0.2s;
+    }
+
+    &:not(:disabled):not(.disabled):active,
+    &:not(:disabled):not(.disabled).active,
+    &.show > .btn-gray.dropdown-toggle {
+      background: $gray-8;
+      box-shadow: inset 0 -0.0625rem 0 0 $gray-8, 0 0 0 0.25rem $rgba-4;
+    }
+
+    &:focus,
+    &.focus {
+      box-shadow: inset 0 -0.0625rem 0 0 $gray-8, 0 0 0 0.25rem $rgba-4;
+    }
+  }
+  // Danger Button
+  &--danger {
+    color: $danger-red-1;
+    background: $white;
+    border-color: $danger-red-1;
+    bottom: 0;
+    transition: background 0.2s, bottom 0.2s;
+
+    &:hover {
+      color: $white;
+      background: $danger-red-1;
+      box-shadow: inset 0 -0.0625rem 0 0 $danger-red-2, 0 0.125rem 0.75rem 0 $rgba-1;
+      bottom: 0.1875rem;
+      transition: background 0.2s, bottom 0.2s;
+    }
+
+    &:not(:disabled):not(.disabled):active,
+    &:not(:disabled):not(.disabled).active,
+    &.show > .btn-gray.dropdown-toggle {
+      background: $danger-red-1;
+      box-shadow: inset 0 -0.0625rem 0 0 $danger-red-1, 0 0 0 0.25rem $rgba-5;
+    }
+
+    &:focus,
+    &.focus {
+      box-shadow: inset 0 -0.0625rem 0 0 $danger-red-1, 0 0 0 0.25rem $rgba-5;
+    }
+  }
+  // Disabled Button - Has to remain at the end to overwrite all hover attributes on defined buttons.
+  &:disabled {
+    background: $white;
+    border: 0.0625rem solid $gray-4;
+    color: $gray-7;
+    opacity: 1;
+    box-shadow: none;
+    transition: initial;
+
+    .msds-icon {
+      fill: $gray-5;
+    }
+
+    &:hover {
+      background: $white;
+      bottom: 0;
+      box-shadow: initial;
+    }
+  }
+}
+
+.msds-btn--icon {
+  padding-right: 1.25rem;
+  padding-left: 1.75rem;
+
+  .msds-icon {
+    vertical-align: bottom;
+    margin-top: 0.0625rem;
+    padding-left: 1.5rem;
+    height: 1.3rem;
+    width: 1.3rem;
+  }
+
+  &.msds-btn--lg {
+    padding: 0.9rem 1.5rem 1.05rem 2.25rem;
+
+    .msds-icon {
+      vertical-align: bottom;
+      margin-top: 0.125rem;
+      padding-left: 4rem;
+    }
+  }
+
+  &.msds-btn--sm {
+    padding-right: 0.75rem;
+    padding-left: 1.25rem;
+
+    .msds-icon {
+      height: 0.9rem;
+      width: 0.9rem;
+      padding-left: 1.555625rem;
+    }
+  }
+
+  &--left {
+    padding-right: 1.75rem;
+    padding-left: 1.25rem;
+
+    .msds-icon {
+      padding-left: 0;
+      padding-right: 1.5rem;
+      margin-top: -0.0625rem;
+    }
+
+    &.msds-btn--lg {
```

Comment:
Modifiers should not be nested but moved up to the same level as other modifiers. Modifiers should work independently of each other. Otherwise, it will be too complicated to understand how it affects styles to the element